src/streams.c

```diff
@@ -1783,11 +1759,11 @@ Obj FuncREAD_ALL_FILE (
           do {
               csize =
                   (ilim == -1 || (ilim - len) > 20000) ? 20000 : ilim - len;
-              lstr = SyRead(ifid, buf, csize);
+              lstr = SyReadWithBuffer(ifid, buf, csize);
           } while (lstr == -1 && errno == EAGAIN);
       }
       if (lstr <= 0) {
-          syBuf[ifid].ateof = 1;
+          SyBufSetEOF(ifid);
```

Comment:
`SyBufSetEOF` is a bit of a hack. It would be better if the code in `sysfiles` took care of setting `ateof`. However, that would have involved modifying not just the new `SyReadWithBuffer ` (which would be OK, I guess), but also `SyFgetsSemiBlock`. I'd prefer to worry about that in a future PR, as this starts to touch some really complicated (if not "messy") code that we want to eventually untangle anyway (e.g. the readline integration)